integration/devmode/devmode_test.go

```diff
@@ -0,0 +1,287 @@
+
+		By("building chaincode")
+		chaincodeExecutePath := components.Build(legacyChaincode.Path)
+
+		By("running the chaincode")
+		legacyChaincodeID := legacyChaincode.Name + ":" + legacyChaincode.Version
+		peerChaincodeAddress := network.PeerAddress(org1peer0, nwo.ChaincodePort)
+		envs := []string{
+			"CORE_PEER_TLS_ENABLED=false",
+			"CORE_CHAINCODE_ID_NAME=" + legacyChaincodeID,
+			"DEVMODE_ENABLED=true",
+		}
+		cmd := exec.Command(chaincodeExecutePath, "-peer.address", peerChaincodeAddress)
+		cmd.Env = append(cmd.Env, envs...)
+		chaincodeRunner = ginkgomon.New(ginkgomon.Config{
+			Name:    "chaincode",
+			Command: cmd,
+		})
+		chaincodeProcess = ifrit.Invoke(chaincodeRunner)
+		Eventually(chaincodeRunner.Buffer()).Should(gbytes.Say("starting up in devmode..."))
+
+		By("installing the chaincode")
+		nwo.InstallChaincodeLegacy(network, legacyChaincode, org1peer0)
+
+		By("instantiating the chaincode")
+		nwo.InstantiateChaincodeLegacy(network, channelName, orderer, legacyChaincode, org1peer0, org1peer0)
+
+		By("querying and invoking the chaincode")
+		RunQueryInvokeQuery(network, orderer, org1peer0, channelName, 100)
+
+		By("killing chaincode process")
+		chaincodeProcess.Signal(syscall.SIGKILL)
+		Eventually(chaincodeProcess.Wait(), network.EventuallyTimeout).Should(Receive())
+
+		By("querying and invoking the chaincode")
+		RunQueryInvokeQuery(network, orderer, org1peer0, channelName, 90)
+
+		Eventually(chaincodeRunner.Buffer()).Should(gbytes.Say("invoking in devmode"))
+	})
+
+	It("executes chaincode in dev mode", func() {
+		chaincode = nwo.Chaincode{
+			Name:            "mycc",
+			Version:         "0.0",
+			Path:            components.Build("github.com/hyperledger/fabric/integration/chaincode/simple/cmd"),
+			Lang:            "binary",
+			PackageFile:     filepath.Join(testDir, "simplecc.tar.gz"),
+			Ctor:            `{"Args":["init","a","100","b","200"]}`,
+			SignaturePolicy: `OR ('Org1MSP.member')`,
+			Sequence:        "1",
+			InitRequired:    true,
+			Label:           "my_prebuilt_chaincode",
+		}
+
+		org1peer0 := network.Peer("Org1", "peer0")
+		orderer := network.Orderer("orderer")
+
+		By("setting up the channel")
+		network.CreateAndJoinChannel(orderer, channelName)
+
+		By("enabling V2_0 application capabilities")
+		nwo.EnableCapabilities(network, channelName, "Application", "V2_0", orderer, org1peer0)
+
+		By("running the chaincode")
+		chaincodeID := chaincode.Name + ":" + chaincode.Version
+		peerChaincodeAddress := network.PeerAddress(org1peer0, nwo.ChaincodePort)
+		envs := []string{
+			"CORE_PEER_TLS_ENABLED=false",
+			"CORE_PEER_ADDRESS=" + peerChaincodeAddress,
+			"CORE_CHAINCODE_ID_NAME=" + chaincodeID,
+		}
+
+		cmd := exec.Command(chaincode.Path, "-peer.address", peerChaincodeAddress)
+		cmd.Env = append(cmd.Env, envs...)
+		chaincodeRunner = ginkgomon.New(ginkgomon.Config{
+			Name:              "chaincode",
+			Command:           cmd,
+			StartCheckTimeout: 15 * time.Second,
```

Comment:
StartCheck (and related chaincode update) coming soon?

Reply:
🧹